glottolog3/models.py

```diff
@@ -541,6 +541,43 @@ class LanguoidSchema(Schema):
     def make_languoid(self, data):
         return Languoid(**data)
 
+def validate_id_type(type):
+    VALID_TYPES = [
+        IdentifierType.iso.value,
+        IdentifierType.glottolog.value,
+        IdentifierType.wals.value,
+        IdentifierType.ethnologue.value,
+        'name'
+    ]
+
+    if not type in VALID_TYPES:
+        raise ValidationError('Invalid identifier type: {}'.format(type))
+
+class IdentifierSchema(Schema):
+    id = fields.Str(default = '')
```

Comment:
do we not want to require an id here?

Reply:
the identifier id isn't something the user provides - it's calculated when they pass in the other fields, so i format it in pre_load